presto-main/src/main/java/io/prestosql/sql/planner/optimizations/ExpressionEquivalence.java

```diff
@@ -285,8 +284,13 @@ public int compare(RowExpression left, RowExpression right)
                 if (javaType == float.class || javaType == double.class) {
                     return Double.compare(((Number) leftValue).doubleValue(), ((Number) rightValue).doubleValue());
                 }
-                if (javaType == Slice.class) {
-                    return ((Slice) leftValue).compareTo((Slice) rightValue);
+                if (leftValue instanceof Comparable) {
+                    try {
+                        //noinspection unchecked
+                        return ((Comparable<Object>) leftValue).compareTo(rightValue);
```

Comment:
techncally this cast is wrong, `leftValue` is not `Comparable<Object>`
i'd rather cast to a raw type `Comparable` (and add `rawtypes` to noinspection as well)